File: backends/go-gin/internal/domain/staff/entity.go
package staff

import "time"

// Staff はスタッフのドメインエンティティです（GORM タグなし）。
type Staff struct {
	ID          uint
	Name        string
	Email       string
	Provider    int
	ProviderID  string
	Avatar      *string
	Role        int
	LastLoginAt *time.Time
	CreatedAt   time.Time
	CreatedBy   *uint
	UpdatedAt   time.Time
	UpdatedBy   *uint
	DeletedAt   *time.Time
	DeletedBy   *uint
	Version     int
}

File: backends/kotlin-ktor/src/main/kotlin/com/authorization/domain/staff/Entity.kt
package com.authorization.domain.staff

import java.time.LocalDateTime

data class Staff(
    val id:          Long           = 0,
    val name:        String,
    val email:       String,
    val provider:    Int,
    val providerId:  String,
    val avatar:      String?        = null,
    val role:        Int            = StaffRole.MEMBER,
    val lastLoginAt: LocalDateTime? = null,
    val createdAt:   LocalDateTime  = LocalDateTime.now(),
    val createdBy:   Long?          = null,
    val updatedAt:   LocalDateTime  = LocalDateTime.now(),
    val updatedBy:   Long?          = null,
    val deletedAt:   LocalDateTime? = null,
    val deletedBy:   Long?          = null,
    val version:     Int            = 0,
)
